src/main/java/br/edu/ifsp/ifspcodelab/gestaoestagiosbackend/curriculum/CurriculumServiceImpl.java

```diff
@@ -122,4 +130,84 @@ private Curriculum getCurriculum(UUID courseId, UUID curriculumId) {
         return curriculumRepository.findAllByCourseIdAndId(courseId, curriculumId)
             .orElseThrow(() -> new ResourceNotFoundException(ResourceName.CURRICULUM, curriculumId));
     }
+
+
+    private void verifyDatesChronology(LocalDate startDate, LocalDate endDate){
+        if(endDate.isBefore(startDate)){
+            throw new DateChronologyException(startDate, endDate);
+        }
+    }
+
+    private void verifyDatesBeforeCreateCurriculum(LocalDate startDate, LocalDate endDate, UUID courseId){
+        List<Curriculum> existedCurriculums = curriculumRepository.findAllByCourseId(courseId);
+        if(endDate != null){
+            if(!existedCurriculums.isEmpty()){
+                verifyDatesChronology(startDate,endDate);
```

Comment:
Esse método pode ser chamado logo no início do método, dessa forma caso a data inicial for maior que a final a exception será lançada já parando a execução do método.